Packet++/header/HttpLayer.h

```diff
@@ -207,200 +207,264 @@ namespace pcpp
+			HttpStatus3xxCodeUnknown = 900003, // 3xx: Redirection - Further action must be taken in order to complete the request
+			HttpStatus4xxCodeUnknown = 900004, // 4xx: Client Error - The request contains bad syntax or cannot be fulfilled
+			HttpStatus5xxCodeUnknown = 900005, // 5xx: Server Error - The server failed to fulfill an apparently valid request
+			HttpStatusCodeError = 999999, // other arbitrary number
+		};
+
+		HttpResponseStatusCode() = default;
+		constexpr HttpResponseStatusCode(Value statusCode) : m_value(statusCode) { }
+
+		constexpr operator Value() const { return m_value; }
+		explicit operator bool() const = delete;
+
+		struct HttpResponseStatusCodeHash {
+			size_t operator()(const HttpResponseStatusCode& status) const {
+				return std::hash<int>()(static_cast<int>(status));
+			}
 		};
 
+		std::string toString() const {
+			return std::to_string(m_value);
+		}
+
+		int toInt() const {
+			return static_cast<int>(m_value);
+		}
+
+		bool isUnsupportedCode() const {
+			return m_value > 599; // any unknown or error code has an extreme large enum value
+		}
+
+		constexpr bool operator==(const HttpResponseStatusCode &other) const { return m_value == other.m_value; }
+		constexpr bool operator!=(const HttpResponseStatusCode &other) const { return m_value != other.m_value; }
+
+		constexpr bool operator==(const Value &otherValue) const { return m_value == otherValue; }
+		constexpr bool operator!=(const Value &otherValue) const { return m_value != otherValue; }
+	private:
+  		Value m_value;
+	};
+
+	// -------- Class HttpResponseLayer -----------------
+
+	class HttpResponseFirstLine;
+
+
+	/**
+	 * @class HttpResponseLayer
+	 * Represents an HTTP response header and inherits all basic functionality of HttpMessage and TextBasedProtocolMessage.
+	 * The functionality that is added for this class is the HTTP first line concept. An HTTP response has the following first line:
+	 * <i>200 OK HTTP/1.1</i>
+	 * Since it's not an "ordinary" HTTP field, it requires a special treatment and gets a class of it's own: HttpResponseFirstLine.
+	 * Unlike most L2-4 protocols, an HTTP response header can spread over more than 1 packet. PcapPlusPlus currently doesn't support a header
+	 * that is spread over more than 1 packet so in such cases: 1) only the first packet will be parsed as HttpResponseLayer (the other packets
+	 * won't be recognized as HttpResponseLayer) and 2) the HTTP header for the first packet won't be complete (as it continues in the following
+	 * packets), this why PcapPlusPlus can indicate that HTTP response header is complete or not (doesn't end with "\r\n\r\n" or "\n\n") using
+	 * HttpMessage#isHeaderComplete()
+	 */
+	class HttpResponseLayer : public HttpMessage
+	{
+		friend class HttpResponseFirstLine;
+	public:
+		// backward compatibility
+		using HttpResponseStatusCode = pcpp::HttpResponseStatusCode;
```

Comment:
i think it's better to put `HttpResponseStatusCode` outside, since it's big, and it's not a good idea to be a subclass.
however, here we still have backward compatibility